src/frontend/js/components/SaleTunnel/index.tsx

```diff
@@ -86,6 +92,15 @@ const SaleTunnel = ({ product, courseCode }: Props) => {
   const productQuery = useProduct(product.id);
   const ordersQuery = useOrders({ product: product.id, course: courseCode });
   const [isOpen, setIsOpen] = useState(false);
+
+  const hasCourseRun = useMemo(() => {
+    let count = 0;
+    product.target_courses.forEach((course) => {
+      count += course.course_runs.length;
+    });
+    return count > 0;
```

Comment:
Block the sell when one target course of the product has no course runs.
```suggestion
return product.target_courses.some(({ course_runs }) => course_runs.length === 0);
```

Reply:
Done